addons/network/XEH_preInit.sqf

```diff
@@ -0,0 +1,56 @@
+#include "script_component.hpp"
+
+ADDON = false;
+
+// Restore loadouts lost by the naked unit bug
+[QGVAR(restoreLoadout), {
+    params ["_unit", "_loadout"];
+    if (count _loadout < 7) exitWith {};
+
+    // Check if loadout got lost
+    private _loadoutLost = false;
+    {
+        (_loadout select _forEachIndex) params [["_backup", "", [""]]];
+        // Reject backup if it contains less information
+        if (_backup isEqualTo "" && !(_x isEqualTo "")) exitWith {};
+        if !(_x isEqualTo _backup) exitWith {
+            _loadoutLost = true;
+        };
+    } forEach [
+        primaryWeapon _unit,
+        secondaryWeapon _unit,
+        handgunWeapon _unit,
+        uniform _unit,
+        vest _unit,
+        backpack _unit,
+        headgear _unit
+    ];
+
+    if (_loadoutLost) then {
+        _unit setUnitLoadout _loadout;
+    };
+}] call CBA_fnc_addEventHandler;
+
+["CAManBase", "Local", {
+    params ["_unit", "_local"];
+    if (_local) then {
+        // Enable loadout restoration for units transferred to the machine
+        _unit setVariable [QGVAR(enableRestoreLoadout), true];
+    } else {
+        // Broadcast loadout to new owner if local was not triggered by init
+        if (_unit getVariable [QGVAR(enableRestoreLoadout), false]) then {
+            private _loadout = getUnitLoadout _unit;
+            [QGVAR(restoreLoadout), [_unit, _loadout], _unit] call CBA_fnc_targetEvent;
```

Comment:
Shouldn't `enableRestoreLoadout` be reset to `false` here as well? Isn't the meaning of the variable just `isLocal` / `wasLocal`?

Reply:
Maybe I haven't described the problem well. When a unit is created, the `local` EH gets triggered on all machines where the unit is not local. This trigger is also added to the jip queue, which explains the issue that @PabstMirror has described. So we essentially only need to suppress the `local` EH on all machines where the unit is not local at the beginning.